neqo-transport/src/connection.rs

```diff
@@ -1633,7 +1633,13 @@ impl Connection {
             }
 
             dump_packet(self, "TX ->", pt, pn, &builder[payload_start..]);
-            qlog::packet_sent(&mut self.qlog, pt, pn, &builder[payload_start..]);
+            qlog::packet_sent(
+                &mut self.qlog,
+                pt,
+                pn,
+                builder.len(),
```

Comment:
This is going to be wrong in a coalesced packet.  You need to subtract payload_start.